src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -909,63 +909,61 @@ impl LinkCollector<'_, '_> {
             (parts[0], None)
         };
         let resolved_self;
-        let link_text;
-        let mut path_str;
         let disambiguator;
-        let (mut res, mut fragment) = {
-            path_str = if let Ok((d, path)) = Disambiguator::from_str(&link) {
-                disambiguator = Some(d);
-                path
-            } else {
-                disambiguator = None;
-                &link
-            }
-            .trim();
 
-            if path_str.contains(|ch: char| !(ch.is_alphanumeric() || ch == ':' || ch == '_')) {
-                return None;
-            }
+        let mut path_str = if let Ok((d, path)) = Disambiguator::from_str(&link) {
```

Comment:
Can you split this refactor into a separate commit so the logic changes are easier to see?